src/main/java/com/google/sps/FakeTextDetectionAPIImpl.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.sps.data.ShoppingListTextEntry;
+import java.util.List;
+
+/**
+  * This is a fake class used in the unit tests to avoid triggering cloudVisionAPI calls.
+  * The test's setup should provide either of the two values
+  * 1) Set the exception status
+  * 2) Set the data value with expected response to be returned back.
+  */
+public class FakeTextDetectionAPIImpl implements TextDetectionAPI {
```

Comment:
Could you add a public default constructor to do some initialization?

e.g. initialize "detectData"

If that's not done, calling "detect" will return null.